yarn-project/pxe/src/storage/note_data_provider/note_data_provider.ts

```diff
@@ -126,7 +189,27 @@ export class NoteDataProvider implements DataProvider {
     });
   }
 
-  public async unnullifyNotesAfter(blockNumber: number, synchedBlockNumber?: number): Promise<void> {
+  /**
+   * Restores notes that were nullified after the given block number.
+   *
+   * This operation "unnullifies" notes, effectively saying "oops... these nullifications
+   * shouldn't count... roll them back!" This is used during blockchain reorganizations
+   * when nullifications that occurred in orphaned blocks need to be undone.
+   *
+   * Effects:
+   * - Finds all nullifiers that occurred after the specified block number
+   * - Retrieves the corresponding nullified notes
+   * - Moves notes from #nullifiedNotes back to #notes (active notes)
+   * - Restores nullifier-to-noteId mappings
+   * - Re-adds notes to all scope-specific indexes
+   * - Cleans up nullified note indexes and mappings
+   * - Removes nullifiers from block number tracking
+   *
+   * @param blockNumber - Nullifications after this block will be reverted
+   * @param synchedBlockNumber - Optional upper bound for the block range to process
+   * @returns Promise that resolves when all notes have been restored
+   */
+  public async restoreNotesAfterBlock(blockNumber: number, synchedBlockNumber?: number): Promise<void> {
```

Comment:
I had to do a double-take as I was confused by this name, particularly due to seeing `deleteNotesAfter` next to `restoreNotesAfter`. I'd consider something like `removeNullifiers` possibly?

Ultimately the store does these things, with the goal of having 'a list of notes':
 - process new notes, adding them to the list
 - process nullifiers that correspond to notes it knows of (it cannot process unknown nullifs), removing notes from the list and putting them in the nullified list
 - un-process nullifiers, restoring notes to the list
 - un-process notes, deleting them outright

'process' is an extremely generic verb so I'm not necessarily saying we should use it, but perhaps it'd make sense to use consistent naming for these actions.